centrallix/utility/obfuscate.c

```diff
@@ -1200,18 +1201,24 @@ obfObfuscateData(pObjData srcval, pObjData dstval, int data_type, char* attrname
 		break;
 
 	    case DATA_T_MONEY:
-		iv = srcval->Money->WholePart * 100 + (srcval->Money->FractionPart / 100);
-		if (strchr(param,'i'))
-		    dv = obf_internal_ObfuscateIntegerMultiples(hash, hash_novalue, &bitcnt, iv);
-		else
-		    {
-		    if (obf_internal_GetBits(hash, &bitcnt, 1)) iv = (iv + 1)*10;
-		    if (obf_internal_GetBits(hash, &bitcnt, 1)) iv = iv/10 + 1;
-		    dv = obf_internal_ObfuscateInteger(hash, hash_novalue, &bitcnt, iv);
-		    }
-		m.WholePart = floor(dv/100.0);
-		m.FractionPart = (dv - m.WholePart*100) * 100;
-		dstval->Money = &m;
+	        if (srcval->Money->Value < INT_MAX)
```

Comment:
Check for negatives, also the max value to check here is: Value < INT_MAX / 100.

Reply:
I will place a check for Value to be above INT_MIN/100 as well.  Is that what you mean by check for negatives?

Reply:
Check to ensure the resulting value can fit in an int32, both at the positive and negative end of the range.

Reply:
I added the check at the beginning to see if srcval->Money->Value is greater than INT_MIN.  By resulting value, do you mean the value placed into m.Value at the end after the obfuscation process, dv?

Reply:
No, the intermediate int32 value that gets passed to the obfuscation routine.

Reply:
Since iv is set to srcval->Money->Value * 100 and I check if srcval->Money->Value is within int max and min /100, isn't the intermediate value already screened to be within an int32?

Reply:
Yes, the main point is to avoid overflow going into iv, not to constrain what the final m.Value is.